controllers/profiles/reconciler_prod.go

```diff
@@ -111,36 +120,58 @@ func (h *newBuilderReconciliationState) Do(ctx context.Context, workflow *operat
 	_, err := platform.GetActivePlatform(ctx, h.client, workflow.Namespace)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			workflow.Status.Manager().MarkFalse(api.RunningConditionType, api.WaitingForPlatformReason,
-				"No active Platform for namespace %s so the workflow cannot be built.", workflow.Namespace)
+			// Recovery isn't possible just signal on events and on log
+			msg := "No active Platform for namespace %s so the workflow cannot be built."
+			workflow.Status.Manager().MarkFalse(api.RunningConditionType, api.WaitingForPlatformReason, msg, workflow.Namespace)
+			h.GenerateErrorEvent(workflow, "SonataFlowPlatformError", fmt.Sprintf(msg, workflow.Namespace))
 			_, err = h.performStatusUpdate(ctx, workflow)
 			return ctrl.Result{RequeueAfter: requeueWhileWaitForPlatform}, nil, err
 		}
+		// Recovery isn't possible just signal on events and on log
 		h.logger.Error(err, "Failed to get active platform")
+		h.GenerateErrorEvent(workflow, "SonataFlowStatusUpdateError", fmt.Sprintf("Failed to get active platform, error: %v", err))
 		return ctrl.Result{RequeueAfter: requeueWhileWaitForPlatform}, nil, err
 	}
 	// If there is an active platform we have got all the information to build but...
 	// ...let's check before if we have got already a build!
 	buildManager := builder.NewSonataFlowBuildManager(ctx, h.client)
 	build, err := buildManager.GetOrCreateBuild(workflow)
 	if err != nil {
+		h.GenerateErrorEvent(workflow, "SonataFlowGetOrCreateBuildError", fmt.Sprintf("Error: %v", err))
 		return ctrl.Result{}, nil, err
 	}
 
 	if build.Status.BuildPhase != operatorapi.BuildPhaseFailed {
 		workflow.Status.Manager().MarkFalse(api.BuiltConditionType, api.BuildIsRunningReason, "")
 		workflow.Status.Manager().MarkFalse(api.RunningConditionType, api.WaitingForBuildReason, "")
 		_, err = h.performStatusUpdate(ctx, workflow)
+		if err != nil {
+			h.GenerateErrorEvent(workflow, "SonataFlowStatusUpdateError", fmt.Sprintf("Error: %v", err))
+		}
 	} else {
-		// TODO: not ideal, but we will improve it on https://issues.redhat.com/browse/KOGITO-8792
-		h.logger.Info("Build is in failed state, try to delete the SonataFlowBuild to restart a new build cycle")
+		msg := "Build is in failed state, try to delete the SonataFlowBuild to restart a new build cycle"
+		h.logger.Info(msg)
+		h.GenerateErrorEvent(workflow, "SonataFlowBuildError", msg)
 	}
 
 	return ctrl.Result{RequeueAfter: requeueAfterStartingBuild}, nil, err
 }
 
 type followBuildStatusReconciliationState struct {
 	*stateSupport
+	recorder record.EventRecorder
+}
+
+func (h *followBuildStatusReconciliationState) GenerateErrorEvent(object runtime.Object, reason string, msg string) {
```

Comment:
isn't better to have just one func where, instead have one func for each event, you pass the event as parameter?>

Reply:
Yep, we can starts with generics in our code